bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/profiles/StateProfileType.java

```diff
@@ -12,15 +12,25 @@
  */
 package org.eclipse.smarthome.core.thing.profiles;
 
+import java.util.Collection;
+
 import org.eclipse.jdt.annotation.NonNullByDefault;
 
 /**
  * Describes a {@link StateProfile} type.
  *
  * @author Simon Kaufmann - initial contribution and API.
+ * @author Stefan Triller - added getSupportedItemTypesOfChannel method
  *
  */
 @NonNullByDefault
 public interface StateProfileType extends ProfileType {
 
+    /**
+     * Get a collection of ItemType names that a Channel needs to support in order to able to use this ProfileType
+     *
+     * @return a collection of supported ItemType names (an empty list means ALL types are supported)
+     */
+    Collection<String> getSupportedItemTypesOfChannel();
```

Comment:
Shouldn't that method be used at any place or do we introduce it for future use only?

Reply:
It is used by #6025